base/server/python/pki/server/deployment/scriptlets/configuration.py

```diff
@@ -654,6 +655,31 @@ def spawn(self, deployer):
 
                 self.validate_system_certs(deployer, nssdb, subsystem)
 
+            elif len(subsystems) > 1:
+
+                for s in subsystems:
+
+                    if s.name == subsystem.name:
```

Comment:
Question: I may be confused with this code. What is the significance of this if condition?

Reply:
We're only installing one subsystem at a time. The `subsystem` points to the current subsystem, and the `s` points to a subsystem that is already installed. The code is basically trying to find a subsystem that is already installed to copy the cert/request data from. Let me add some comments into the code.

This is actually adapted from an existing code that is no longer executed due to this bug:
https://github.com/dogtagpki/pki/blob/master/base/server/python/pki/server/deployment/pkihelper.py#L2934-L2942
The new code is basically replacing the old code, we can clean up the old code later.